cmake/protobuf.cmake

```diff
@@ -73,10 +77,16 @@ elseif(gRPC_PROTOBUF_PROVIDER STREQUAL "package")
     endif()
     if(TARGET protobuf::protoc)
       set(_gRPC_PROTOBUF_PROTOC protobuf::protoc)
-      set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE $<TARGET_FILE:protobuf::protoc>)
+      if(CMAKE_CROSSCOMPILING)
+        find_program(_gRPC_PROTOBUF_PROTOC_EXECUTABLE protoc)
+      else()
+        set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE $<TARGET_FILE:protobuf::protoc>)
+      endif()
     else()
       set(_gRPC_PROTOBUF_PROTOC ${PROTOBUF_PROTOC_EXECUTABLE})
-      set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE ${PROTOBUF_PROTOC_EXECUTABLE})
+      if(NOT DEFINED _gRPC_PROTOBUF_PROTOC_EXECUTABLE)
+        set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE ${PROTOBUF_PROTOC_EXECUTABLE})
```

Comment:
as said in the other comment, checking if _gRPC_PROTOBUF_PROTOC_EXECUTABLE is already set feels inconsistent (because in other places the variable is just overwritten, so it creates a false impression that it's fair to set this variable). 
Is there any reason why the usual `if(CMAKE_CROSSCOMPILING)` and `find_program()` wouldn't work here?

Reply:
Nope, just an oversight on my part. Sorry about that, thanks for catching it.